apps/hpc-ftsadmin/src/app/pages/flows/pending-flows-list.tsx

```diff
@@ -0,0 +1,129 @@
+import { C, CLASSES, combineClasses } from '@unocha/hpc-ui';
+import { t } from '../../../i18n';
+import PageMeta from '../../components/page-meta';
+import { AppContext } from '../../context';
+import {
+  JsonParam,
+  NumberParam,
+  StringParam,
+  createEnumParam,
+  decodeNumber,
+  useQueryParams,
+  withDefault,
+} from 'use-query-params';
+import FilterPendingFlowsTable, {
+  PENDING_FLOWS_FILTER_INITIAL_VALUES,
+} from '../../components/filters/filter-pending-flows-table';
+import tw from 'twin.macro';
+import {
+  DEFAULT_FLOW_TABLE_HEADERS,
+  encodeTableHeaders,
+} from '../../utils/table-headers';
+import FlowsTable, {
+  FlowsTableProps,
+} from '../../components/tables/flows-table';
+import { useCallback, useEffect, useState } from 'react';
+
+interface Props {
+  className?: string;
+}
+const Container = tw.div`
+flex
+`;
+const LandingContainer = tw.div`
+w-full
+`;
+
+export default (props: Props) => {
+  const [abortController, setAbortController] = useState<AbortController>(
+    new AbortController()
+  );
+
+  const [query, setQuery] = useQueryParams({
+    page: withDefault(NumberParam, 0),
+    rowsPerPage: withDefault(
+      {
+        ...NumberParam,
+        decode: (string) => {
+          // prevent user requesting more than max number of rows
+          const number = decodeNumber(string);
+          return number && Math.min(number, Math.max(...[10, 25, 50, 100]));
+        },
+      },
+      50
+    ),
+    orderBy: withDefault(
+      createEnumParam(
+        // Same as filter then map but this is acceptable to typescript
+        DEFAULT_FLOW_TABLE_HEADERS.reduce((acc, curr) => {
+          if (curr.sortable) {
+            return [...acc, curr.identifierID];
+          }
+
+          return acc;
+        }, [] as string[])
+      ),
+      'flow.updatedAt'
+    ),
+    orderDir: withDefault(createEnumParam(['ASC', 'DESC']), 'DESC'),
+    filters: withDefault(JsonParam, JSON.stringify({})),
+    tableHeaders: withDefault(StringParam, encodeTableHeaders([])), //  Default value of table headers
+    prevPageCursor: withDefault(NumberParam, 0),
+    nextPageCursor: withDefault(NumberParam, 0),
+  });
+
+  const handleAbortController = useCallback(() => {
+    // Abort the ongoing requests
+    abortController.abort();
+
+    // Create a new AbortController for the next requests
+    const newAbortController = new AbortController();
+    setAbortController(newAbortController);
+
+    // Perform actions with the updated filter values
+
+    // Pass the new AbortSignal to FlowsTableGraphQL
+    // This can be part of your state or directly passed as a prop
+  }, [abortController]);
+
+  useEffect(() => {
+    return () => {
+      handleAbortController();
+    };
+  }, []);
```

Comment:
So, you decided to ignore it 🫣

Reply:
Okay so for this, it may be a lack of knowledge from my part in react, but basically, I want to just execute when the component gets unmounted.  My inner guts tells me this is not the way to go, because `useEffect` in general shouldn't be used for this cases but its in my head of old ways to use react when people used classes to be able to define what to do `onUnmount`.

If I write handleAbortController it will update if handleAbortController changes (what it shouldn't) but in any case I just want this to be updated onUnmount, and it could be missleading to say that is depending on change of `handleAbortController`

What do you think is the real proper solution, because this warning even if I solve it in other places, it just gives me problems.

Reply:
Honestly, linter warnings around the dependencies of `useEffect` are often giving me problems too. I don't always know how to resolve it, but I am not too skilled with frontend development and React.

In this case, I extracted the code pieces from this component and asked Claude AI models for help. After some back & forth, it offered me what sounds a sensible solution, but take it with a grain of salt, because AI models hallucinate too often.

The main point is that your `handleAbortController` is aborting the request and then creating a new `AbortController`. But, when component is unmounting, you just need to abort the request, there is no point in creating a new `AbortController`. So, if we just call `abortController.abort()` in the `useEffect` handler, instead of entire `handleAbortController`, that should be enough.

AI bot also suggested switching from `useState` to `useRef`, presumably because we would not need to add it to dependencies of `useEffect`, but I don't know, because it doesn't warn me on local and I cannot even find annotations in the GitHub Actions runs that I wrote about originally. Something may be broken with linting configuration.

Anyway, this is the code it produced.
```tsx
const abortControllerRef = useRef<AbortController>(new AbortController());

// const [query, setQuery] = useQueryParams({ ... });

const handleAbortController = useCallback(() => {
  abortControllerRef.current.abort();
  abortControllerRef.current = new AbortController();
}, []);

useEffect(() => {
  return () => {
    abortControllerRef.current.abort();
  };
}, []);

const pendingFlowsTableProps: FlowsTableProps = {
  headers: DEFAULT_FLOW_TABLE_HEADERS,
  initialValues: PENDING_FLOWS_FILTER_INITIAL_VALUES,
  rowsPerPageOption: [10, 25, 50, 100],
  query: query,
  setQuery: setQuery,
  pending: true,
  abortSignal: abortControllerRef.current.signal,
};
```